data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/LocationData.java

```diff
@@ -1,43 +1,164 @@
+    @JsonProperty(CITY_FIELD)
     private final String cityName;
+    @JsonProperty(LOCATION_FIELD)
+    private final Map<String, Double> location;
+    @JsonProperty(LATITUDE_FIELD)
+    private final Double latitude;
+    @JsonProperty(LONGITUDE_FIELD)
+    private final Double longitude;
+    @JsonProperty(CONTINENT_FIELD)
+    private final String continentCode;
+    @JsonProperty(COUNTRY_ISO_FIELD)
+    private final String countryCode;
+    @JsonProperty(POSTAL_FIELD)
+    private final String postalCode;
+    @JsonProperty(REGION_CODE_FIELD)
+    private final String regionCode;
+    @JsonProperty(TIMEZONE_FIELD)
+    private final String timeZone;
 
-    public LocationData(String countryName, String subdivisionName, String cityName) {
-        this.countryName = countryName;
-        this.subdivisionName = subdivisionName;
-        this.cityName = cityName;
-    }
+    public LocationData(Builder builder) {
+        ip = builder.ip;
+        countryName = builder.countryName;
+        regionName = builder.regionName;
+        cityName = builder.cityName;
+        Double[] latAndLong = builder.location;
+        continentCode = builder.continentCode;
+        countryCode = builder.countryCode;
+        postalCode = builder.postalCode;
+        regionCode = builder.regionCode;
+        timeZone = builder.timeZone;
 
-    @Override
-    public String toString() {
-        return "LocationData{" +
-                "countryName='" + countryName + '\'' +
-                ", subdivisionName='" + subdivisionName + '\'' +
-                ", cityName='" + cityName + '\'' +
-                '}';
+        if (latAndLong != null && latAndLong.length == 2) {
+            location = new HashMap<String, Double>() {{
+                put("lat", latAndLong[0]);
+                put("lon", latAndLong[1]);
+            }};
+        } else {
+            location = null;
+        }
+        latitude = builder.latitude;
+        longitude = builder.longitude;
     }
-    //TODO Update this to use ObjectMapper to format the json data.
-
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         LocationData that = (LocationData) o;
-        return Objects.equals(countryName, that.countryName) && Objects.equals(subdivisionName, that.subdivisionName) && Objects.equals(cityName, that.cityName);
+        return Objects.equals(ip, that.ip) && Objects.equals(countryName, that.countryName) && Objects.equals(regionName, that.regionName) && Objects.equals(cityName, that.cityName) && Objects.equals(location, that.location) && Objects.equals(latitude, that.latitude) && Objects.equals(longitude, that.longitude) && Objects.equals(continentCode, that.continentCode) && Objects.equals(countryCode, that.countryCode) && Objects.equals(postalCode, that.postalCode) && Objects.equals(regionCode, that.regionCode) && Objects.equals(timeZone, that.timeZone);
```

Comment:
Surprised this didn't get formatted into line-breaks. Nit - would suggest multi-lines, starting with the `&&` on each line.